lib/ansible/modules/system/service.py

```diff
@@ -1157,7 +1157,7 @@ def get_service_status(self):
 
     def service_control(self):
         if self.enable_cmd:
-            return self.execute_command("%s -f %s %s" % (self.svc_cmd, self.action, self.name))
+            return self.execute_command("%s -f %s %s" % (self.svc_cmd, self.action, self.name), daemonize=True)
```

Comment:
we had this before and had to remove as this would always 'succeed' while services would fail to start up, so if we have to do it for 'bg' services, it should be conditional and only applied to those.

Reply:
Thanks for the feedback.

Could you point me to a specific commit or issue? I cannot reproduce this behaviour. With the changes above in place, failing services seem to be recognised as such for me:

Failing `rc_bg=YES` service:
```
~/scratch/ansible-50460 % ansible temp -i hosts -u root -m service -a 'name=ypbind state=started'
temp01 | FAILED! => {
    "changed": false,
    "msg": "ypbind(failed)\n"
}
```

Failing `rc_bg=NO` service:
```
~/scratch/ansible-50460 % ansible temp -i hosts -u root -m service -a 'name=bgpd state=started'
temp01 | FAILED! => {
    "changed": false,
    "msg": "bgpd(failed)\n"
}
```

Both of these can be reproduced with a pristine OpenBSD target system after installing Python (`ansible temp -i hosts -u root -m raw -a 'pkg_add python%3'`).

Am I missing something here?

Reply:
IIRC the original issue was with tomcat

Reply:
Do you remember any additional details?

I cannot reproduce any problems using tomcat and `daemonize=True` that do not occur using `daemonize=False` or `rcctl start tomcat` locally.

Reply:
sadly no, it was 'irc report' and we (mostly I) were not good at updating ticktes with all the info, since you cannot reproduce, I'll assume its a non issue now and will wait for someone to complain to then have to revisit this

Reply:
Sounds good, thanks for your help!